backend/server/src/main.rs

```diff
@@ -32,7 +32,12 @@ async fn main() -> Result<()> {
 
     match args.cmd {
         None => start_server(&args).await?,
-        Some(Command::Config { target }) => config::write_template(target.as_ref())?,
+        Some(Command::Config { target }) => {
```

Comment:
This seems very boilerplate-y. Is there no way to restrict options to subcommands? (And have a certain subcommand be the default, I guess)?

Reply:
I assume you meant to comment on the `if` below? Restricting options to sub-commands is easy. However, specifying that an option is forbidden for a specific subcommand is not possible, unfortunately :/ And it's also not possible to specify options that are only valid if no subcommand is specified. I searched for such a feature, but didn't find anything.

Reply:
You assume correctly. What is it with me putting comments in the wrong place in this PR?! :sweat_smile: 

Regarding the issue: Command line parsing libraries continue to disappoint me even in the Rust ecosystem, I see. :disappointed:

Reply:
This is unfortunate, but I very much like `structopt`. Just for the protocol ;-)